pkg/skaffold/runner/runner.go

```diff
@@ -170,13 +170,28 @@ func getTester(cfg []*latest.TestCase, opts *config.SkaffoldOptions) (test.Teste
 	}
 }
 
+// kubeContext returns the kubectl context to use for deployment
+func getContext(originalContext, schemaContext string) string {
+	if kc := os.Getenv("KUBE_CONTEXT"); kc != "" {
+		return kc
+	}
+
+	if kc := schemaContext; kc != "" {
+		return kc
+	}
+
+	return originalContext
+}
+
 func getDeployer(cfg *latest.DeployConfig, kubeContext string, namespace string, defaultRepo string) (deploy.Deployer, error) {
 	// TODO(dgageot): this should be the folder containing skaffold.yaml. Should also be moved elsewhere.
 	cwd, err := os.Getwd()
 	if err != nil {
 		return nil, errors.Wrap(err, "finding current directory")
 	}
 
+	kubeContext = getContext(kubeContext, cfg.KubeContext)
```

Comment:
As already pointed out in the previous revision, `getContext` should be called one level up in 
`NewForConfig` (pkg/skaffold/runner/runner.go:64) because actual context also needs to be passed to `getBuilder` (pkg/skaffold/runner/runner.go:86).
Also, `GetLocalCluster` (cmd/skaffold/app/cmd/config/util.go:177) needs to know about the actual kubecontext in order to determine whether to treat the build as local or not. For that, you need to pass in `kubeContext` as an additional argument to `GetLocalCluster`.

Reply:
Apologies - I'd missed that. `GetLocalCluster` now accepts kubeContext.